cmd/d2/main.go

```diff
@@ -165,12 +180,19 @@ func compile(ctx context.Context, ms *xmain.State, plugin d2plugin.Plugin, input
 	if err != nil {
 		return nil, err
 	}
-	svg, err = plugin.PostProcess(ctx, svg)
+	outputImage, err := plugin.PostProcess(ctx, svg)
 	if err != nil {
 		return nil, err
 	}
 
-	err = ms.WritePath(outputPath, svg)
+	if filepath.Ext(outputPath) == ".png" {
+		outputImage, err = png.ConvertSVG(ms, page, outputImage)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	err = ms.WritePath(outputPath, outputImage)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
i just noticed that below this line, ur still returning `svg`, which is wrong now.

Reply:
So this is svg because the browser only accepts SVGs right now as a render.
```
			w.broadcast(&compileResult{
				SVG: string(b),
			})
```

Should we make it so that it can broadcast PNGs?

Reply:
no, the `svg` variable used to also include the postprocess. u changed that but didn't change the return